src/main/scala/it/pagopa/interop/notifier/service/impl/QueueHandler.scala

```diff
@@ -24,50 +23,58 @@ final class QueueHandler(
+  private val logger: Logger                 = Logger(this.getClass)
 
-  /*
-   * it does:
-   *
-   *  - M2M token generation for calling services downstream
-   *  - given a message payload, retrieves the recipient of the message
-   *  - gets the next event id for the corresponding recipient
-   *  - saves the event on dynamo
-   */
-  def processMessage(msg: Message)(implicit scanamo: ScanamoAsync): Future[Unit] = for {
-    m2mToken <- interopTokenGenerator
-      .generateInternalToken(
-        subject = jwtConfig.subject,
-        audience = jwtConfig.audience.toList,
-        tokenIssuer = jwtConfig.issuer,
-        secondsDuration = jwtConfig.durationInSeconds
-      )
-    m2mContexts = Seq(CORRELATION_ID_HEADER -> UUID.randomUUID().toString, BEARER -> m2mToken.serialized)
-    messageId <- extractMessageId(msg.payload, dynamoIndexService)(scanamo, m2mContexts)
-    _ = logger.debug(s"Organization id retrieved for message  ${msg.messageUUID} -> ${messageId.organizationId}")
-    nextEvent <- idRetriever.getNextEventIdForOrganization(messageId.organizationId)(m2mContexts)
-    _ = logger.debug(s"Next event id for organization ${nextEvent.organizationId} -> ${nextEvent.eventId}")
-    notificationMessage <- NotificationMessage.create(messageId, nextEvent.eventId, msg).toFuture
-    result              <- dynamoNotificationService.put(notificationMessage)
-    _ = logger.debug(s"Message ${msg.messageUUID.toString} was successfully written to dynamodb")
-  } yield result
+  private def generateM2mContexts: Future[List[(String, String)]] = interopTokenGenerator
+    .generateInternalToken(
+      subject = jwtConfig.subject,
+      audience = jwtConfig.audience.toList,
+      tokenIssuer = jwtConfig.issuer,
+      secondsDuration = jwtConfig.durationInSeconds
+    )
+    .map(m2mToken => List(CORRELATION_ID_HEADER -> UUID.randomUUID().toString, BEARER -> m2mToken.serialized))
 
-  private[this] def extractMessageId(
-    event: ProjectableEvent,
-    dynamoIndexService: DynamoNotificationResourcesService
-  )(implicit scanamo: ScanamoAsync, contexts: Seq[(String, String)]): Future[MessageId] = {
-    val composed: PartialFunction[ProjectableEvent, Future[MessageId]] = {
-      PurposeEventsConverter.getMessageId(catalogManagementService, dynamoIndexService) orElse
-        AgreementEventsConverter.getMessageId(dynamoIndexService) orElse
-        CatalogEventsConverter.getMessageId(dynamoIndexService) orElse
-        notFoundRecipient
-    }
+  def processMessage(msg: Message): Future[Unit] = generateM2mContexts.flatMap(messageFlow(_, msg))
 
-    composed(event)
+  private def messageFlow(contexts: Seq[(String, String)], message: Message): Future[Unit] =
+    dynamoFlow(contexts)
+      .orElse(postgreFlow(contexts))
+      .applyOrElse(message, notFoundRecipient)
+
+  private def postgreFlow: Seq[(String, String)] => PartialFunction[Message, Future[Unit]] = _ => { m: Message =>
+    authorizationEventsHandler.handleEvents(m)
   }
```

Comment:
If the context is not used
```suggestion
  private def postgreFlow: PartialFunction[Message, Future[Unit]] = authorizationEventsHandler.handleEvents(m)
```